vici/requests.go

```diff
@@ -62,6 +62,9 @@ func (s *Session) sendStreamedRequest(cmd string, event string, msg *Message) ([
 }
 
 func (s *Session) handleStreamedRequest(cmd, event string, msg *Message) ([]*Message, error) {
+	if s.ctr == nil {
```

Comment:
Is there any reason not to do this check immediately after acquiring the lock? I.e. in `sendStreamedRequest`?

Reply:
Yes, I want to do this check right in the method where is `s.ctr` used. You know, in the future you can have more "starting points" and the issue can happen again.

Reply:
The `sendRequest` and `sendStreamedRequest` functions act as entry points for use of `s.ctr`, because they acquire the lock. I.e. any other functions that use `s.ctr` should not be called from outside a path that started with `sendRequest` or `sendStreamedRequest`.

I think it is clearer to check `s.ctr` immediately after acquiring the lock.

Reply:
Besides, we should be able to tell right then and there if we're going to be able to use `s.ctr`. So what's the point of executing more code just to bail out later?

Reply:
ok, fixed

Reply:
Did you forget to push? I don't see any changes.

Reply:
forget to add changed file 🤦‍♀️